sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -780,7 +780,7 @@ class SparkSession private(
     DataSource.lookupDataSource(runner, sessionState.conf) match {
       case source if classOf[ExternalCommandRunner].isAssignableFrom(source) =>
         Dataset.ofRows(self, ExternalCommandExecutor(
-          source.getDeclaredConstructor().newInstance()
+          DataSource.newDataSourceInstance(runner, source)
```

Comment:
It may be arguable that if this is a breaking change. Now people need to worry about python data source in the code that is to deal with DS v1 only.

Reply:
`ExternalCommandRunner` is DSv2 API..